cypress/components/button-bar/button-bar.cy.tsx

```diff
@@ -148,27 +153,72 @@ context("Test for Button-Bar component", () => {
     });
   });
 
-  it("has the expected border radius styling when major button children passed", () => {
-    CypressMountWithProviders(<ButtonBarCustom />);
-
-    buttonDataComponent()
-      .eq(0)
-      .should("have.css", "border-radius", "32px 0px 0px 32px");
-    buttonDataComponent().eq(1).should("have.css", "border-radius", "0px");
-    buttonDataComponent()
-      .eq(2)
-      .should("have.css", "border-radius", "0px 32px 32px 0px");
-  });
-
-  it("has the expected border radius styling when minor button children passed", () => {
-    CypressMountWithProviders(<ButtonBarWithMinorButtonChildren />);
-
-    buttonDataComponent()
-      .eq(0)
-      .should("have.css", "border-radius", "4px 0px 0px 4px");
-    buttonDataComponent().eq(1).should("have.css", "border-radius", "0px");
-    buttonDataComponent()
-      .eq(2)
-      .should("have.css", "border-radius", "0px 4px 4px 0px");
+  describe("renders with ButtonMinor children", () => {
+    const buttonTypesAndBackgrounds = [
+      ["1st", "primary", 0, "rgb(162, 44, 59)"],
+      ["2nd", "secondary", 1, "rgba(0, 0, 0, 0)"],
+      ["3rd", "tertiary", 2, "rgba(0, 0, 0, 0)"],
+    ];
+
+    describe("renders with ButtonMinor children", () => {
+      it.each(buttonTypesAndBackgrounds)(
+        "should apply correct background-color on hover for %s button rendered with ButtonMinor children",
+
+// https://github.com/bahmutov/cypress-each/issues/2
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        (_: any, buttonType: ButtonTypes, index: number, result: string) => {
+          CypressMountWithProviders(<ButtonBarMinor buttonType={buttonType} />);
+
+          buttonMinorComponent(index).should(
+            "have.css",
+            "background-color",
+            result
+          );
+          buttonMinorComponent(index).realHover();
+
+          // reset focus
+          cyRoot().realHover({ position: "topLeft" });
+        }
+      );
+
+      it.each(buttonTypesAndBackgrounds)(
+        "should check Button Minor Bar have correct color for the button at index %n",
```

Comment:
![image](https://github.com/Sage/carbon/assets/44157880/3fc0db2f-fe59-43b4-98a0-9a6625b90c34)
 This outputs the above which isn't very useful, I think something like the suggestion below will be better. You have also re-used the same `buttonTypesAndBackgrounds` here but will need to create a new `buttonTypesAndColors` array with the correct values for these tests to pass
 
 ```suggestion
        "should apply the correct color to the %s button when %s ButtonMinor children passed",
```